### PR TITLE
USWDS - Changelogs: Fix changelog dates

### LINKED ISSUE #2199: USWDS - Bug: Fix dates in Changelog
There are changelog entries that show `NNNN-NN-NN` instead of the correct date.

- Identifier [`changelogs/component-identifier.yml`]
- Table [`changelogs/component-table.yml`]
- Utilities [`changelogs/utilities.yml`]

### PR DESCRIPTION
**Pull request by mejiaj**
# Summary

Replace placeholder dates with actual dates.

## Related issue

Closes #2199.

## Preview link

| Page | Old | New |
|:--------|------|------|
| Identifier | [preview](https://designsystem.digital.gov/components/identifier/#latest-updates) | [preview](https://federalist-ead78f8d-8948-417c-a957-c21ec5617a57.sites.pages.cloud.gov/preview/uswds/uswds-site/jm-bugfix-changelog-dates/components/identifier/#latest-updates) |
| Table | [preview](https://designsystem.digital.gov/components/table/#latest-updates) | [preview](https://federalist-ead78f8d-8948-417c-a957-c21ec5617a57.sites.pages.cloud.gov/preview/uswds/uswds-site/jm-bugfix-changelog-dates/components/table/#latest-updates) |
| Utilities | [preview](https://designsystem.digital.gov/utilities/#latest-updates) | [preview](https://federalist-ead78f8d-8948-417c-a957-c21ec5617a57.sites.pages.cloud.gov/preview/uswds/uswds-site/jm-bugfix-changelog-dates/utilities/#latest-updates) | 

## Problem statement

Changelog dates were missing.

## Solution

Added date based on when PR was originally merged.

## Major changes

_For complex PRs, create a list of the significant updates made._

## Testing and review

Changelog entries should have real dates.

### CODE REVIEW

File: _data/changelogs/component-table.yml [hunk 1]
```diff
@@ -4,7 +4,7 @@ type: component
 changelogURL:
 
 items:
-  - date: NNNN-NN-NN
+  - date: 2023-06-17
```
Comment (amyleadem) [inline _data/changelogs/component-table.yml:7]: Is this the accurate date for the merge? The PR shows it was merged [two days ago](https://github.com/uswds/uswds-site/pull/2143#event-9838545393).
```suggestion
  - date: 2023-07-17
```

File: _data/changelogs/utilities.yml [hunk 2]
```diff
@@ -9,7 +9,7 @@ items:
     githubPr: 5209
     githubRepo: uswds
     versionUswds: 3.5.0
-  - date: NNN-NN-NN
+  - date: 2023-05-26
```
Comment (amyleadem) [inline _data/changelogs/utilities.yml:12]: Should these be the same release date as other 3.5.0 issues? [3.5.0](https://github.com/uswds/uswds/releases/tag/v3.5.0) was released on June 9. 
```suggestion
  - date: 2023-06-09
```

**Review by amyleadem — CHANGES_REQUESTED** [review]
Thanks for catching these! 
I added a couple of questions in the comments. Let me know if you have questions or comments.

File: _data/changelogs/component-identifier.yml [hunk 3]
```diff
@@ -2,21 +2,21 @@ title: Identifier
...
-    affectsGuidance: true
-    githubPr: 2097
-    githubRepo: uswds-site
-    versionUswds: 3.5.0
   - date: 2023-06-09
-    summary: Updated Accessibility statement link text. 
+    summary: Updated Accessibility statement link text.
     summaryAdditional: Updated the identifier to use the text "Accessibility statement" (EN) / "Declaración de accesibilidad" (ES) for the required link to an accessibility statement.
     isBreaking: true
     affectsContent: true
```
[the remark this answers is not in the document]
Comment (mejiaj) [reply _data/changelogs/component-identifier.yml:9]: Which tag? Let's create a new issue for it. I'd like this PR scoped to fixing the dates.

**Comment by mejiaj** [comment]
@amyleadem thanks for the suggestions. Re-requested review 😄

File: _data/changelogs/component-identifier.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (amyleadem) [reply _data/changelogs/component-identifier.yml:9]: Opened https://github.com/uswds/uswds-site/issues/2202 to address this

**Review by amyleadem — APPROVED** [review]
Thanks, @mejiaj!

**Pull request merged**